### PR TITLE
API split

### PR DESCRIPTION
**Pull request by mbouron-gpsw**

### CODE REVIEW

File: libnodegl/backends/gl/api_gl.c [hunk 1]
```diff
@@ -0,0 +1,140 @@
+#include "config.h"
```
Comment (cboesch-gpsw) [inline libnodegl/backends/gl/api_gl.c:1]: missing copyright header

File: libnodegl/backends/gl/api_gl.c [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+#include "backends/gl/gpu_ctx_gl.h"
```
Comment (cboesch-gpsw) [inline libnodegl/backends/gl/api_gl.c:3]: This one is not needed anymore in `api.c`

File: libnodegl/backends/vk/api_vk.c [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#include "internal.h"
```
Comment (cboesch-gpsw) [inline libnodegl/backends/vk/api_vk.c:1]: missing copyright header

File: libnodegl/backends/gl/api_gl.c [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+}
+
+static int gl_set_capture_buffer(struct ngl_ctx *s, void *capture_buffer)
+{
+    return ngli_ctx_dispatch_cmd(s, cmd_set_capture_buffer, capture_buffer);
+}
+
+static int cmd_set_scene(struct ngl_ctx *s, void *arg)
+{
+    ngli_gpu_ctx_wait_idle(s->gpu_ctx);
```
Comment (cboesch-gpsw) [inline libnodegl/backends/gl/api_gl.c:88]: vk doesn't need this one?

File: libnodegl/backends/gl/api_gl.c [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+        cmd_make_current(s, MAKE_CURRENT);
+        ret = ngli_ctx_resize(s, width, height, viewport);
+        if (ret < 0)
+            return ret;
+        cmd_make_current(s, DONE_CURRENT);
+
+        return ngli_ctx_dispatch_cmd(s, cmd_make_current, MAKE_CURRENT);
+    }
+
+    struct resize_params params = {
```
Comment (cboesch-gpsw) [inline libnodegl/backends/gl/api_gl.c:68]: Do you mind marking it const and casting the dispatch call? The struct is actually const, only the transmission layer needs the cast hack.

Edit: OK just realized it preserves the current code, so feel free to ignore or change it in a separate commit.

File: libnodegl/backends/gl/gpu_ctx_gl.h [hunk 6]
```diff
@@ -74,4 +74,7 @@ struct gpu_ctx_gl {
     void (*glGetQueryObjectui64v)(const struct glcontext *gl, GLuint id, GLenum pname, GLuint64 *params);
 };
 
+int ngli_gpu_ctx_gl_make_current(struct gpu_ctx *s);
+int ngli_gpu_ctx_gl_release_current(struct gpu_ctx *s);
```
Comment (cboesch-gpsw) [inline libnodegl/backends/gl/gpu_ctx_gl.h:78]: Are you going to add other functions like this by removing some very GL oriented callbacks in `struct gpu_ctx_class` at some point?

**Review by cboesch-gpsw — CHANGES_REQUESTED** [review]
Nice.

File: libnodegl/backends/gl/gpu_ctx_gl.h (hunk 6 above)
Comment (mbouron-gpsw) [reply libnodegl/backends/gl/gpu_ctx_gl.h:78]: Yes, typically `ngli_gpu_ctx_gl_reset_state(struct gpu_ctx *s);` that will come with the upcoming PR adding support for external OpenGL contexts.

File: libnodegl/backends/gl/api_gl.c (hunk 1 above)
Comment (mbouron-gpsw) [reply libnodegl/backends/gl/api_gl.c:22]: Done.

File: libnodegl/backends/gl/api_gl.c (hunk 2 above)
Comment (mbouron-gpsw) [reply libnodegl/backends/gl/api_gl.c:24]: Done.

File: libnodegl/backends/vk/api_vk.c (hunk 3 above)
Comment (mbouron-gpsw) [reply libnodegl/backends/vk/api_vk.c:22]: Done.

File: libnodegl/backends/gl/api_gl.c (hunk 4 above)
Comment (mbouron-gpsw) [reply libnodegl/backends/gl/api_gl.c]: Indeed it does, I moved `ngli_gpu_ctx_wait_idle(s->gpu_ctx);` to `ngli_ctx_set_scene()` in a separate commit.

**Review by cboesch-gpsw — APPROVED** [review]

**Pull request merged**